src/robotFactory.js

```diff
@@ -1,10 +1,80 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords = {
+    x: 0, y: 0,
+  }, chipVersion) {
```

Comment:
Let's make your code cleaner
```suggestion
  constructor(name, weight, coords = {
    x: 0,
    y: 0
  }, chipVersion) {
```